Source/Common/Config.cpp

```diff
@@ -70,6 +74,65 @@ namespace FEX::Config {
 
       std::string Program = Args[0];
 
+      // If executed with a FEX FD then the Program argument might be empty.
+      // In this case we need to scan the FD node to recover the application binary that exists on disk.
+      // Only do this if the Program argument is empty, since we would prefer the application's expectation
+      // of application name.
+      if (ProgramFDFromEnv && Program.empty()) {
+        // Get the execvefd string, then link to the `dev` node of that FD.
+        char *FDString = getenv("FEX_EXECVEFD");
+
+        if (FDString) {
+          Program = "/dev/fd/";
+          Program += std::string(FDString);
+        }
+      }
+
+      // If FEX was invoked through an FD path (either binfmt_misc or execveat) then we need to check the
+      // Program to see if it is a symlink to find the real path.
+      //
+      // binfmt_misc: Arg[0] is actually the full application path or `/dev/fd/<FD>` path
+      //   - Full application path with execve (See Side Note)
+      //   - FD path with execveat and FD doesn't have an existing filesystem on the disk
+      //
+      // ProgramFDFromEnv: Arg[0] is set to Application provided data or `/dev/fd/<FD>` from above fix-up.
```

Comment:
This is useful information, but it mixes "things that applications do" with "how they do it" and (maybe?) "how FEX handles" it. Currently it's difficult to tell which of these you're describing at which point.

If you're describing what values certain values take, I'd use "the value of X will be" over "X is set", since the latter is ambiguous about who sets it (FEX before this line? FEX after this line? The application? ...).

If you're describing values that we actively set after the comment, active phrasing ("We set"/"FEX sets") helps clarify this over passive phrasing ("is set").